diff -urN a/cpp/src/arrow/compute/exec/ir_consumer.cc b/cpp/src/arrow/compute/exec/ir_consumer.cc
--- a/cpp/src/arrow/compute/exec/ir_consumer.cc	2022-05-03 19:59:12.000000000 +0300
+++ b/cpp/src/arrow/compute/exec/ir_consumer.cc	2022-06-30 14:49:52.869775774 +0300
@@ -26,6 +26,7 @@
 #include "arrow/compute/function_internal.h"
 #include "arrow/ipc/dictionary.h"
 #include "arrow/ipc/metadata_internal.h"
+#include "arrow/util/tracing_internal.h"
 #include "arrow/util/unreachable.h"
 #include "arrow/visit_type_inline.h"
 
diff -urN a/cpp/src/arrow/compute/exec/options.cc b/cpp/src/arrow/compute/exec/options.cc
--- a/cpp/src/arrow/compute/exec/options.cc	2022-05-03 19:59:12.000000000 +0300
+++ b/cpp/src/arrow/compute/exec/options.cc	2022-06-30 15:05:41.723158702 +0300
@@ -21,6 +21,7 @@
 #include "arrow/table.h"
 #include "arrow/util/async_generator.h"
 #include "arrow/util/logging.h"
+#include "arrow/util/tracing_internal.h"
 
 namespace arrow {
 namespace compute {
diff -urN a/cpp/src/arrow/compute/exec/tpch_node.cc b/cpp/src/arrow/compute/exec/tpch_node.cc
--- a/cpp/src/arrow/compute/exec/tpch_node.cc	2022-05-03 19:59:12.000000000 +0300
+++ b/cpp/src/arrow/compute/exec/tpch_node.cc	2022-06-30 15:18:29.140693152 +0300
@@ -23,6 +23,7 @@
 #include "arrow/util/io_util.h"
 #include "arrow/util/make_unique.h"
 #include "arrow/util/pcg_random.h"
+#include "arrow/util/tracing_internal.h"
 #include "arrow/util/unreachable.h"
 
 #include <algorithm>
diff -urN a/cpp/src/arrow/compute/exec/util.cc b/cpp/src/arrow/compute/exec/util.cc
--- a/cpp/src/arrow/compute/exec/util.cc	2022-05-03 19:59:12.000000000 +0300
+++ b/cpp/src/arrow/compute/exec/util.cc	2022-06-30 15:18:36.768855040 +0300
@@ -21,6 +21,7 @@
 #include "arrow/table.h"
 #include "arrow/util/bit_util.h"
 #include "arrow/util/bitmap_ops.h"
+#include "arrow/util/tracing_internal.h"
 #include "arrow/util/ubsan.h"
 
 namespace arrow {
